pkg/proxy/proxy.go

```diff
@@ -0,0 +1,319 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cilium/cilium/pkg/policy"
+
+	"github.com/braintree/manners"
+	"github.com/op/go-logging"
+	"github.com/spf13/viper"
+	"github.com/vulcand/oxy/forward"
+	"github.com/vulcand/route"
+)
+
+var (
+	log     = logging.MustGetLogger("cilium-proxy")
+	logFile *os.File
+	logBuf  *bufio.Writer
+)
+
+type Redirect struct {
+	id       string
+	FromPort uint16
+	ToPort   uint16
+	Rules    []policy.AuxRule
+	source   ProxySource
+	server   *manners.GracefulServer
+	router   route.Router
+}
+
+func (r *Redirect) updateRules(rules []policy.AuxRule) {
+	for _, v := range r.Rules {
+		r.router.RemoveRoute(v.Expr)
+	}
+
+	r.Rules = make([]policy.AuxRule, len(rules))
+	copy(r.Rules, rules)
+
+	for _, v := range r.Rules {
+		r.router.AddRoute(v.Expr, v)
+	}
+}
+
+type ProxySource interface {
+}
+
+type Proxy struct {
+	rangeMin       uint16
```

Comment:
This is only the proxy port allocation which should be invisible to the user. It does not have any impact the ports we are redirecting from.